File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
from collections import deque
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root is None:
            return []
        result=[]
        queue=deque()
        queue.append(root)
        while queue:
            currentLevelSize=len(queue)
            currentLevel=[]
            for _ in range(currentLevelSize):
                currentNode=queue.popleft()
                currentLevel.append(currentNode.val)
                if currentNode.left:
                    queue.append(currentNode.left)
                if currentNode.right:
                    queue.append(currentNode.right)
            result.append(currentLevel)
        return result

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result = new ArrayList<>();
        if(root==null){
            return result;
        }
        Deque<TreeNode> deque=new ArrayDeque<>();
        deque.offer(root);
        int currentQueueLength;
        List<Integer> currentLevel;
        TreeNode currentNode;
        while(!deque.isEmpty()){
            currentQueueLength=deque.size();
            currentLevel=new ArrayList<Integer>();
            for(int i=0;i<currentQueueLength;i++){
                currentNode=deque.poll();
                currentLevel.add(currentNode.val);
                if(currentNode.left!=null){
                    deque.offer(currentNode.left);
                }
                if(currentNode.right!=null){
                    deque.offer(currentNode.right);
                }
            }
            result.add(currentLevel);
        }
        return result;
        
        
    }
}